consensus/qbft/src/main/java/org/hyperledger/besu/consensus/qbft/statemachine/QbftBlockHeightManager.java

```diff
@@ -99,50 +95,66 @@ public QbftBlockHeightManager(
                 finalState.getQuorum(),
                 messageValidatorFactory.createMessageValidator(roundIdentifier, parentHeader));
 
-    currentRound = roundFactory.createNewRound(parentHeader, 0);
-    if (finalState.isLocalNodeProposerForRound(currentRound.getRoundIdentifier())) {
-      blockTimer.startTimer(currentRound.getRoundIdentifier(), parentHeader);
-    }
+    final long nextBlockHeight = parentHeader.getNumber() + 1;
+    final ConsensusRoundIdentifier roundIdentifier =
+        new ConsensusRoundIdentifier(nextBlockHeight, 0);
+
+    finalState.getBlockTimer().startTimer(roundIdentifier, parentHeader);
   }
 
   @Override
   public void handleBlockTimerExpiry(final ConsensusRoundIdentifier roundIdentifier) {
-    if (roundIdentifier.equals(currentRound.getRoundIdentifier())) {
-      currentRound.createAndSendProposalMessage(clock.millis() / 1000L);
+    startNewRound(0);
+    final QbftRound qbftRound = currentRound.get();
+
+    if (!finalState.isLocalNodeProposerForRound(qbftRound.getRoundIdentifier())) {
```

Comment:
We must check that the node is not the proposer for round 0, not for the current round.

Reply:
The current round is 0 at this point. As initialise the round above with startNewRound(0). I've added a comment and verify in the tests to show that this is the case. Also have this returning immediately is the round has already started before we receive the block expiry.